### PR TITLE
2095 undefined classes for nesy patterns

### LINKED ISSUE #2095: Undefined classes for NeSy patterns
```
%prefix( : <http://www.ontohub.org/meta/> )%
logic NeSyPatterns    
pattern Model = data NeSyPatterns.omn
  Model;
end
```
gives
```
Analyzing file file:///home/till/temp/nesy.dol as library nesy
Downloading NeSyPatterns.omn ...
Analyzing library http://www.ontohub.org/meta/NeSyPatterns
logic OWL
Analyzing spec http://www.ontohub.org/meta/NeSyPatterns
... loaded NeSyPatterns.omn
logic NeSyPatterns
Analyzing spec Model
file:///home/till/temp/nesy.dol:4.3-5.1: *** Error:
Undefined class 'Model'
hets: user error (Stopped due to errors)
```

### PR DESCRIPTION
**Pull request by b-gehrke**
Closes #2095.

Added expansion for symbols.

### CODE REVIEW

**Comment by tillmo** [comment]
`SYMB` should be based on `node`, `Symbol` be based on `ResolvedNode`.
A `SYMB_MAP_ITEMS` can use just ontology terms, if they lead to unique id. Otherwise, the id has to specified as well. This means that the static analysis of `SYMB_MAP_ITEMS` etc. should get a signature.

**Comment by b-gehrke** [comment]
@tillmo Changes are implemented as discussed. Ready for review.

**Comment by tillmo** [comment]
For the example in the issue, I still get:
```
$ ./hets -g ~/temp/nesy5.dol 
Analyzing file file:///home/till/temp/nesy5.dol as library nesy5
Downloading NeSyPatterns.omn ...
Analyzing library http://www.ontohub.org/meta/NeSyPatterns
logic OWL
Analyzing spec http://www.ontohub.org/meta/NeSyPatterns
... loaded NeSyPatterns.omn
logic NeSyPatterns
Analyzing spec Model
file:///home/till/temp/nesy5.dol:4.3-5.1: *** Error:
Undefined class 'Model'
hets: user error (Stopped due to errors)
```

**Comment by mcodescu** [comment]
With `www` in the empty prefix, I get
```
Analyzing file file:///home/mcodescu/repos/Hets2/nesyView.dol as library nesyView
Downloading ontohub:NeSyPatterns.omn ...
Analyzing library http://www.ontohub.org/meta/NeSyPatterns
logic OWL
Analyzing spec http://www.ontohub.org/meta/NeSyPatterns
... loaded ontohub:NeSyPatterns.omn
logic NeSyPatterns
Analyzing spec Model
file:///home/mcodescu/repos/Hets2/nesyView.dol:4.3-5.1: *** Error:
Undefined class 'gn_nesy0 : Model'
hets: user error (Stopped due to errors)

```

File: NeSyPatterns/Sign.hs [hunk 1]
```diff
@@ -50,6 +50,7 @@ import NeSyPatterns.Print()
 
 import OWL2.AS
 import OWL2.Pretty
+import Debug.Trace (trace)
```
Comment (tillmo) [inline NeSyPatterns/Sign.hs:53]: this should go

File: NeSyPatterns/Sign.hs [hunk 2]
```diff
@@ -174,4 +175,4 @@ sigDiff sig1 sig2 = Sign
 
 {- | union of Signatures, using Result -}
 sigUnion :: Sign -> Sign -> Result Sign
-sigUnion s = return . unite s
+sigUnion s1 s2 = trace ("----sigUnion: s1: " ++ show s1 ++ ", s2: " ++ show s2) $ return $ unite s1 s2
```
Comment (tillmo) [inline NeSyPatterns/Sign.hs:178]: this should go

**Review by tillmo — CHANGES_REQUESTED** [review]
please remove the tracing stuff

**Comment by tillmo** [comment]
> Have you added the `www` to the empty prefix?

oh, that has solved it, thanks.

**Comment by mcodescu** [comment]
> With `www` in the empty prefix, I get
> 
> ```
> Analyzing file file:///home/mcodescu/repos/Hets2/nesyView.dol as library nesyView
> Downloading ontohub:NeSyPatterns.omn ...
> Analyzing library http://www.ontohub.org/meta/NeSyPatterns
> logic OWL
> Analyzing spec http://www.ontohub.org/meta/NeSyPatterns
> ... loaded ontohub:NeSyPatterns.omn
> logic NeSyPatterns
> Analyzing spec Model
> file:///home/mcodescu/repos/Hets2/nesyView.dol:4.3-5.1: *** Error:
> Undefined class 'gn_nesy0 : Model'
> hets: user error (Stopped due to errors)
> ```

I have added some tracing and this is the reason: we compare
```
scheme:file:
authority:IRIAuth {iriUserInfo = "", iriRegName = "nesyView.dol", iriPort = ""}
path:
query:?Model
fragment:
prefix:
iFragment:Model
isAbbrev:True
```
and
```
scheme:http:
authority:IRIAuth {iriUserInfo = "", iriRegName = "www.ontohub.org", iriPort = ""}
path:/meta/Model
query:
fragment:
prefix:
iFragment:Model
isAbbrev:True
```
and for one hasFullIRI holds while for the other it does not. Do you also get the same behavior?

File: NeSyPatterns/Sign.hs (hunk 1 above)
Comment (b-gehrke) [reply NeSyPatterns/Sign.hs]: Removed

File: NeSyPatterns/Sign.hs (hunk 2 above)
Comment (b-gehrke) [reply NeSyPatterns/Sign.hs]: Removed

**Comment by b-gehrke** [comment]
> With `www` in the empty prefix, I get [...]

I don't get these errors. Can you post the contents of `nesyView.dol`?



> I have added some tracing and this is the reason: we compare
> and for one hasFullIRI holds while for the other it does not. Do you also get the same behavior?

Actually, for both `hasFullIRI` holds: The `iriScheme` is not empty in both cases (`file` and `http`). But, indeed, the IRIs are not identical. The absolute IRIs differ (`file://nesyView.dol?Model` and `http://www.ontohub.org/meta/Model`). This should not be as an empty prefix is defined and should overwrite the default default empty prefix.

**Comment by mcodescu** [comment]
Sure, here it is:
```
%prefix( ontohub: <https://www.ontohub.org/meta/> )% 
logic NeSyPatterns    
pattern Model = data ontohub:NeSyPatterns.omn
  Model;
end
```

**Comment by mcodescu** [comment]
> Actually, for both `hasFullIRI` holds

Of course, you're right.

**Comment by mcodescu** [comment]
I get the same error with `:` instead of `ontohub:` and on a fresh copy of the repository on a machine that didn't have Hets installed before.

**Comment by b-gehrke** [comment]
> ```
> %prefix( ontohub: <https://www.ontohub.org/meta/> )% 
>```

The problem is the `https` scheme. In the `NeSyPatterns.omn` the empty prefix is defined with `http` instead of `https`.

The following should go through:

```
%prefix( : <http://www.ontohub.org/meta/> )% 
logic NeSyPatterns    
pattern Model = data :NeSyPatterns.omn
  Model;
end
```


```
%prefix( ontohub: <http://www.ontohub.org/meta/> )% 
logic NeSyPatterns    
pattern Model = data ontohub:NeSyPatterns.omn
  ontohub:Model;
end
```

```
%prefix( : <http://www.ontohub.org/meta/>
  ontohub: <http://www.ontohub.org/meta/> )% 
logic NeSyPatterns    
pattern Model = data ontohub:NeSyPatterns.omn
  Model;
end
```

Maybe we should output a warning if prefixes defined in a data logic conflict with existing prefixes?

**Comment by tillmo** [comment]
I have changed the ontology now:
```
Prefix: : <https://ontohub.org/meta/>`
```
This avoids these problems. `https` has become standard now.

I think it is also a good idea to issue a warning if prefixes conflict.

**Review by tillmo — APPROVED** [review]

**Pull request merged**